### PR TITLE
doc: Documentation of nRF52 Desktop configuration channel

### PR DESCRIPTION
**Pull request by Qbicz**
Added configuration channel API description and documentation page
explaining its behavior.

Jira: DESK-571

Signed-off-by: Filip Kubicz <filip.kubicz@nordicsemi.no>

### CODE REVIEW

**Comment by Qbicz** [comment]
@b-gent, @pdunaj can you review?

**Comment by Qbicz** [comment]
For convenience, see output at hizz/components -> ncs_docs/PR-855/nrf/applications/nrf_desktop/README.html

**Comment by Qbicz** [comment]
@MarekPieta can you review?

File: applications/nrf_desktop/src/util/config_channel.h [hunk 1]
```diff
@@ -7,16 +7,38 @@
...
+/**
+ * @file
+ * @defgroup config_channel Configuration channel
+ * @{
+ * @brief API for the nRF52 Desktop configuration channel.
+ */
+
 #include "config_event.h"
 
+/** Maximum length of the fetched data */
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/util/config_channel.h]: missing dot, maybe remove "the"?

File: applications/nrf_desktop/src/util/config_channel.h [hunk 2]
```diff
@@ -29,34 +51,123 @@ struct fetch {
 	u16_t recipient;
 };
 
+/** @brief Configuration channel instance.
+ */
 struct config_channel_state {
+	/** Status of the current transaction. */
 	atomic_t status;
+
+	/** @c true there is a transaction in progress */
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/util/config_channel.h]: missing if and dot

File: applications/nrf_desktop/src/util/config_channel.h [hunk 3]
```diff
@@ -29,34 +51,123 @@ struct fetch {
...
+ */
 int config_channel_report_parse(const u8_t *buffer, size_t length,
 				struct config_channel_frame *frame, bool usb);
+/**
+ * @brief Fill the configuration channel report with values from provided structure.
+ *
+ * @param buffer Pointer to report buffer which will be filled.
+ * @param length Length of the buffer.
+ * @param frame Pointer to a structure with values which will be copied to buffer.
+ * @param usb @c true If the operation occurs for USB, @c false for Bluetooth.
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/util/config_channel.h]: if, also in other places where parameter usb is described

File: applications/nrf_desktop/src/util/config_channel.h [hunk 4]
```diff
@@ -29,34 +51,123 @@ struct fetch {
...
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/util/config_channel.h]: if, also in other places where returned value is described

File: scripts/hid_configurator/README.rst [hunk 5]
```diff
@@ -1,55 +1,78 @@
...
+
+.. parsed-literal::
+   :class: highlight
 
-On Linux:
+   python3 configurator.py dfu signed-dfu-image.bin
 
-	python3 configurator.py config sensor cpi 12000
-	python3 configurator.py config sensor downshift_rest1 fetch
+To get a list of possible arguments at each subcommand level, call with `--help` option.
```
Comment (MarekPieta) [inline scripts/hid_configurator/README.rst]: after build it shows _-help_, you may want to e.g. escape each character in double dash

File: scripts/hid_configurator/README.rst [hunk 6]
```diff
@@ -1,55 +1,78 @@
...
+   :class: highlight
+
+   python3 configurator.py --help
+   python3 configurator.py config sensor --help
 
-To get a list of possible arguments, call with `--help` option.
 
 Installation
 ************
 To setup dependencies for the configurator program, run:
```
Comment (MarekPieta) [inline scripts/hid_configurator/README.rst]: missing new line (shows "On Windows:" in the same line)

File: scripts/hid_configurator/README.rst [hunk 7]
```diff
@@ -1,55 +1,78 @@
...
 On Windows:
 
-	py -3 -m pip install -r requirements.txt
+.. parsed-literal::
+   :class: highlight
+
+   py -3 -m pip install -r requirements.txt
 
-On Linux:
+On Debian-based Linuxes:
```
Comment (MarekPieta) [inline scripts/hid_configurator/README.rst]: Linux?

File: scripts/hid_configurator/README.rst [hunk 8]
```diff
@@ -63,12 +86,23 @@ On Linux, to call the Python script without root rights,
 install the provided udev rule `99-hid.rules` by copying it to
 `/etc/udev/rules.d` and replugging the device.
 
-Note:
+.. note::
+   When using Python Launcher for Windows, `--user` installation option will not work, since DLL files will be copied to `user\AppData\Roaming\Python\PythonX` which is not searched for DLL libraries by default.
```
Comment (MarekPieta) [inline scripts/hid_configurator/README.rst]: escaping double dash

File: scripts/hid_configurator/README.rst [hunk 9]
```diff
@@ -1,55 +1,78 @@
...
 
-	pip3 install --user -r requirements.txt
-        sudo apt install libhidapi-hidraw0
+.. parsed-literal::
+   :class: highlight
+
+   pip3 install --user -r requirements.txt
+   sudo apt install libhidapi-hidraw0
 
 This installs:
```
Comment (MarekPieta) [inline scripts/hid_configurator/README.rst]: Update for Windows

Comment (Qbicz) [reply scripts/hid_configurator/README.rst]: This is accurate information for now, let's update when we will change the installation method.

**Comment by Qbicz** [comment]
@MarekPieta changes are applied. Also resolved conflicts.

File: scripts/hid_configurator/README.rst (hunk 8 above)
Comment (Qbicz) [reply scripts/hid_configurator/README.rst]: used ``command`` which keeps commands verbatim

**Comment by b-gent** [comment]
I'll finish on Monday

**Comment by Qbicz** [comment]
@b-gent thanks for the review. I agree with the changes. Should I squash and add you as co-author?

**Comment by Qbicz** [comment]
Squashed and rebased. Let's wait for CI.

**Comment by thst-nordic** [comment]
`continuous-integration/jenkins/mcuboot_ctf`   doesn't report to github anymore.   You can ignore this.

**Comment by Qbicz** [comment]
@b-gent can we merge?

**Pull request merged**